### DISCUSSION TITLE
Can we combine reference only control with masactrl?

### DISCUSSION THREAD

**Discussion by xpeng** [Mikubill/sd-webui-controlnet#1311] [Ideas]
now there is a control for consistent Image synthesis and editing named MasaCtrl, which can generate a continuous series of images. it seems the first image generated in this series is most important to content. (https://ljzycmd.github.io/projects/MasaCtrl)

well in img2img, now we have great reference only control mode, may be it can initialize the first image mentioned above for certain purpose perfectly.

however, when i enable reference only mode in control net UI, there comes crash with masactrl webui extension activated.

**Comment by lllyasviel** [comment]
Masked attention link will greatly increase the engineering difficulty in maintaining the robustness of application, but we will consider it once any user can find any real case where masked attention outperforms the simple attention link in reference preprocessors.

DDIM inversion is another story – but considering now users already have many ways to use DDIM inversion in A1111, perhaps implement it in CN again will make things more duplicated. And DDIM inversion has some well-known problems in supporting different samplers and SD2.

**Reply by xpeng** [reply]
i see, thanks

**Reply by lavenderrz** [reply]
Could you elaborate on problems in supporting DDIM for different samplers and SD2?
